### PR TITLE
refactor: allow usage of NiPoint3 and NiQuaternion in constexpr context

### PR DESCRIPTION
**Pull request by jadebenn**
Allows constexpr evaluation of the NiPoint3 and NiQuaternion types, as well as all but one of their member functions (the sole exception was causing a nasty circular dependency issue and probably isn't very useful in a constexpr context anyway).

### CODE REVIEW

File: dCommon/NiPoint3.h
Comment (EmosewaMC) [inline dCommon/NiPoint3.h:179]: yes, it is redundant.  Nothing uses this

**Review by EmosewaMC — DISMISSED** [review]
I have no reason to not be in favor of this.  you convinced me otherwise.

File: dCommon/NiPoint3.h
Comment (jadebenn) [reply dCommon/NiPoint3.h:179]: It is part of the original code though. Maybe there's a reason for it? I'm hesitant to remove it unless there's a pressing need to.

Comment (EmosewaMC) [reply dCommon/NiPoint3.h]: This code is just copied I can almost 100% tell you that and ill show you why in discord

File: dCommon/NiPoint3.inl
Comment (EmosewaMC) [inline dCommon/NiPoint3.inl:1]: pragma once not needed for inl files, if anything the inverse is necessary
```
#ifndef NiPoint3HeaderGuard
#error "This is only to be included inline in NiPoint3.h.  Do not include directly"
#endif
```

Comment (EmosewaMC) [reply dCommon/NiPoint3.inl:1]: Would probably want a combo of the two if you wanted to make sure the inl is also only included once

Comment (jadebenn) [reply dCommon/NiPoint3.inl:1]: Good idea. I'll implement that in a bit.

**Review by aronwk-aaron — APPROVED** [review]

**Review by aronwk-aaron — DISMISSED** [review]

**Review by aronwk-aaron — DISMISSED** [review]

**Review by Xiphoseer — DISMISSED** [review]
Don't quite get why the inl is necessary, otherwise lgtm

**Comment by EmosewaMC** [comment]
> Don't quite get why the inl is necessary, otherwise lgtm

Helps separate out definitions from declarations imo, I would do the same for Entity's template definitions, except in a .tpp file instead of right in the header.

**Review by EmosewaMC — DISMISSED** [review]
lgtm pending passing CI.

**Review by Xiphoseer — DISMISSED** [review]

**Comment by EmosewaMC** [comment]
The static vars need to be compiled once in a cpp file somewhere since they are static and not inline.

**Review by Jettford — APPROVED** [review]

File: .github/workflows/build-and-test.yml
Comment (EmosewaMC) [inline .github/workflows/build-and-test.yml]: remove

File: dCommon/NiPoint3.h
Comment (EmosewaMC) [inline dCommon/NiPoint3.h:44]: Why do you use this syntax for PODs?  To me it adds more clutter and is more cumbersome than just doing this in the constructor body, and since these are POD the gain from doing a chained construction like this is lost since there is no default constructor to not call first.

Comment (jadebenn) [reply dCommon/NiPoint3.h:44]: It's the syntax I'm more used to at this point, but it doesn't make a difference for basic data like floats, yeah.

Comment (jadebenn) [reply dCommon/NiPoint3.h:44]: Also, to restate what I said in the discord, you also get compiler warnings for certain conversions when you do it this way. Not really an issue here, but part of the reason I've gotten into the habit of it.

Comment (EmosewaMC) [reply dCommon/NiPoint3.h:44]: This way being chained constructors or in body construction?

Comment (jadebenn) [reply dCommon/NiPoint3.h:44]: Constructors utilizing member initialization lists in general.

**Comment by jadebenn** [comment]
Can I get some new reviews on this when you all have a chance? It's been ready for a while.

**Review by aronwk-aaron — APPROVED** [review]

**Review by Jettford — APPROVED** [review]

**Pull request merged**
